src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -439,6 +440,35 @@ private void OnTextAlignmentChanged()
 
 #endregion
 
+#region HorizontalTextAlignment Dependency Property
+
+		public new TextAlignment HorizontalTextAlignment
+		{
+			get { return (TextAlignment)this.GetValue(HorizontalTextAlignmentProperty); }
+			set { this.SetValue(HorizontalTextAlignmentProperty, value); }
+		}
+
+		public static readonly DependencyProperty HorizontalTextAlignmentProperty { get; } =
+			DependencyProperty.Register(
+				"HorizontalTextAlignment",
+				typeof(TextAlignment),
+				typeof(TextBlock),
+				new PropertyMetadata(
+					defaultValue: TextAlignment.Left,
+					propertyChangedCallback: (s, e) => ((TextBlock)s).OnHorizontalTextAlignmentChanged()
+				)
+			);
+
+		// This property provides the same functionality as the TextAlignment property.
+		// If both properties are set to conflicting values, the last one set is used.
+		// https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textbox.horizontaltextalignment#remarks
+		private void OnHorizontalTextAlignmentChanged()
+		{
+			TextAlignment = HorizontalTextAlignment;
```

Comment:
What if the TextAlignment was databound? You are breaking the binding here.

Reply:
Good question... but binding both TextAlignment and HorizontalTextAlignment will already produce unexpected values... so it's a theoric issue, not a real one.